### PR TITLE
Improve cross-links in Blazor server-side host doc

### LINKED ISSUE #12445: No info on how to do Blazor server side with Azure SignalR service
This page is pretty low on info in general. It would be great to have info here on the changes I need to make for my server side Blazor app to run on Azure (or another host). In particular, I want my app to work 100% locally when I do dev, and use Azure SignalR when I deploy to SignalR. This includes doing secure credential storage and all the best practices.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 042b0c63-cbb8-0347-97f2-11e935aae405
* Version Independent ID: 064606f3-3a82-1493-768a-41e056eb92de
* Content: [Host and deploy Blazor server-side](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/blazor/server-side?view=aspnetcore-3.0)
* Content Source: [aspnetcore/host-and-deploy/blazor/server-side.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/host-and-deploy/blazor/server-side.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-hostdeploy**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #12445

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/blazor/server-side?view=aspnetcore-3.0&branch=pr-en-us-12732)

Improve cross-linking to content that pertains to hosing server-side Blazor.

### CODE REVIEW

**Comment by bradygaster** [comment]
[this one](https://review.docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new?branch=pr-en-us-12732) 404'ed for me as did [this one](https://review.docs.microsoft.com/en-us/azure/azure-signalr/?branch=pr-en-us-12732), and [this one](https://review.docs.microsoft.com/en-us/azure/azure-signalr/signalr-quickstart-dotnet-core?branch=pr-en-us-12732). 

Aside from those dead links, which probably are only dead b/c I'm in review mode, this is **great and much-appreciated**.

**Comment by guardrex** [comment]
@bradygaster 

> this one 404'ed for me

Our relative links don't work on preview. All of the review topic links end up `https://review.docs.microsoft.com/.....`. The "review" added there breaks all of the links. It's normal.

**Comment by bradygaster** [comment]
yes i know i just needed to call it out because ODC. :)

**Comment by guardrex** [comment]
For those following along, "ODC" refers to [Ordinary Decent Criminal](https://en.wikipedia.org/wiki/Ordinary_Decent_Criminal_(slang)). We're just trying to keep the Foundation afloat here 💰.

**Comment by guardrex** [comment]
oy @bradygaster ... It's a true honor and pleasure work'in with you. 🍻

**Comment by danroth27** [comment]
@guardrex I've looked through the proposed links and I think we need to say a bit more here about the need for a connection scale out solution for server-side Blazor apps. Instead of the "SignalR and Azure resources" section I think there should be a section called "Connection scale out" where we document that server-side Blazor apps require an active SignalR connection per user. Any production server-side Blazor deployment will need a solution for supporting as many concurrent connections as required by the app. Scaling up to lots of concurrent connection can be a difficult problem. That's what the Azure SignalR Service does for you and we strongly recommend using it.

Most of the Azure SignalR Service docs assume that the user is setting up SignalR themselves, but with server-side Blazor that's done for you. The hub used by Blazor is an implementation detail of the framework. So I think we need to explain this before sending folks off to the Azure SignalR Service docs to figure out which parts are relevant to them. Or we could just document the steps ourselves: 1. Provision the Azure SignalR Service, 2. Add the Azure SignalR Service package ([Microsoft.Azure.SignalR](https://nuget.org/packages/Microsoft.Azure.SignalR)), 2. Configure the connection string. 

The next update to Visual Studio will do these steps for you when adding an Azure SignalR Service dependency to your app on publish. VS will also detect that your app is using SignalR/Blazor and recommend that you add the Azure SignalR Service dependency.

Eventually, we will also need to dramatically expand this topic to cover the various aspects of hosting server-side Blazor apps in production, like dealing with state persistence, but we should tackle that later in the Preview7 milestone.

**Comment by bradygaster** [comment]
I do think it'd be great to keep the links outbound to the D&R and scale-out articles. Folks ask about those two things **a lot**, so if we pre-load those links up front it'd be good.

**Comment by guardrex** [comment]
I can finish this up after I work **Add Azure SignalR Service to the SignalR publish-to-Azure page** (#12744). It's in discussion at ...

https://github.com/aspnet/AspNetCore.Docs/issues/12744#issuecomment-499964746

Once that content is in place (and I plan to work on that this weekend), I'll work this PR forward.

**Comment by guardrex** [comment]
@danroth27 I wordsmithed your lingo into two sections:

* Azure SignalR Service with a cross-link.
* Note that SignalR docs are generally for custom and advanced scenarios.

File: aspnetcore/host-and-deploy/blazor/server-side.md [hunk 1]
```diff
@@ -22,19 +22,21 @@ With the [server-side hosting model](xref:blazor/hosting-models#server-side), Bl
...
 A web server capable of hosting an ASP.NET Core app is required. Visual Studio includes the **Blazor (server-side)** project template (`blazorserverside` template when using the [dotnet new](/dotnet/core/tools/dotnet-new) command).
 
-<!--
+## Connection scale out
 
-**INSERT: Concerns are the same as publishing an ASP.NET Core SignalR app**
+Blazor server-side apps require one active SignalR connection for each user. A production Blazor server-side deployment requires a solution for supporting as many concurrent connections as required by the app. The Azure SignalR Service handles scaling connections and is recommended as a scaling solution when deploying to Azure App Service. For more information, see <xref:signalr/publish-to-azure-web-app>.
 
-**INSERT: Content on the Azure SignalR Service**
+In a future release of Visual Studio, you'll have the option to add an Azure SignalR Service dependency to an app on publish.
```
Comment (danroth27) [inline aspnetcore/host-and-deploy/blazor/server-side.md]: I think you can remove this line

File: aspnetcore/host-and-deploy/blazor/server-side.md [hunk 2]
```diff
@@ -22,19 +22,21 @@ With the [server-side hosting model](xref:blazor/hosting-models#server-side), Bl
 
 A web server capable of hosting an ASP.NET Core app is required. Visual Studio includes the **Blazor (server-side)** project template (`blazorserverside` template when using the [dotnet new](/dotnet/core/tools/dotnet-new) command).
 
-<!--
+## Connection scale out
 
-**INSERT: Concerns are the same as publishing an ASP.NET Core SignalR app**
+Blazor server-side apps require one active SignalR connection for each user. A production Blazor server-side deployment requires a solution for supporting as many concurrent connections as required by the app. The Azure SignalR Service handles scaling connections and is recommended as a scaling solution when deploying to Azure App Service. For more information, see <xref:signalr/publish-to-azure-web-app>.
```
Comment (danroth27) [inline aspnetcore/host-and-deploy/blazor/server-side.md]:
```suggestion
Blazor server-side apps require one active SignalR connection for each user. A production Blazor server-side deployment requires a solution for supporting as many concurrent connections as required by the app. The [Azure SignalR Service](/azure/azure-signalr/) handles scaling connections and is recommended as a scaling solution for server-side Blazor apps. For more information, see <xref:signalr/publish-to-azure-web-app>.
```

**Comment by guardrex** [comment]
I'll merge this after the SignalR topic updates go in on https://github.com/aspnet/AspNetCore.Docs/pull/12764.

@danroth27 Note that there's an open question in my opening comments of that PR about how to handle the VS versioning. Seems like that has to be addressed before it can go forward.

**Comment by guardrex** [comment]
@Rick-Anderson This was just on hold until #12764 merged, and that just happened. The changes here refer to updated coverage in that topic on that PR. This is ready for :rocket: *LAUNCH!* :rocket:

**Pull request merged**